File: tests-playwright/pages/checkout-page.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './base-page';

export class CheckoutPage extends BasePage {
  private readonly checkoutContainer: Locator;
  private readonly firstNameInput: Locator;
  private readonly lastNameInput: Locator;
  private readonly postalCodeInput: Locator;
  private readonly continueButton: Locator;
  private readonly cancelButton: Locator;
  private readonly errorMessage: Locator;

  constructor(page: Page) {
    super(page);
    this.checkoutContainer = page.locator('#checkout_info_container');
    this.firstNameInput = page.locator('[data-test="firstName"]');
    this.lastNameInput = page.locator('[data-test="lastName"]');
    this.postalCodeInput = page.locator('[data-test="postalCode"]');
    this.continueButton = page.locator('[data-test="continue"]');
    this.cancelButton = page.locator('[data-test="cancel"]');
    this.errorMessage = page.locator('[data-test="error"]');
  }

  async fillCheckoutInfo(firstName: string, lastName: string, postalCode: string): Promise<void> {
    await this.fillInput(this.firstNameInput, firstName);
    await this.fillInput(this.lastNameInput, lastName);
    await this.fillInput(this.postalCodeInput, postalCode);
  }

  async continueToOverview(): Promise<void> {
    await this.clickElement(this.continueButton);
  }

  async cancelCheckout(): Promise<void> {
    await this.clickElement(this.cancelButton);
  }

  async getErrorMessage(): Promise<string> {
    return await this.getElementText(this.errorMessage);
  }

  async isErrorMessageVisible(): Promise<boolean> {
    return await this.isElementVisible(this.errorMessage);
  }

  async isCheckoutPageLoaded(): Promise<boolean> {
    return await this.isElementVisible(this.checkoutContainer);
  }

  async clearForm(): Promise<void> {
    await this.firstNameInput.clear();
    await this.lastNameInput.clear();
    await this.postalCodeInput.clear();
  }
}

File: tests-csharp/Pages/CheckoutPage.cs
using Microsoft.Playwright;
using Tests.CSharp.Pages;

namespace Tests.CSharp.Pages;

public class CheckoutPage : BasePage
{
    private readonly ILocator _firstNameInput;
    private readonly ILocator _lastNameInput;
    private readonly ILocator _postalCodeInput;
    private readonly ILocator _continueButton;
    private readonly ILocator _cancelButton;
    private readonly ILocator _checkoutContainer;

    public CheckoutPage(IPage page) : base(page)
    {
        _firstNameInput = Page.Locator("[data-test='firstName']");
        _lastNameInput = Page.Locator("[data-test='lastName']");
        _postalCodeInput = Page.Locator("[data-test='postalCode']");
        _continueButton = Page.Locator("[data-test='continue']");
        _cancelButton = Page.Locator("[data-test='cancel']");
        _checkoutContainer = Page.Locator("#checkout_info_container");
    }

    public async Task FillCheckoutInfoAsync(string firstName, string lastName, string postalCode)
    {
        await FillElementAsync(_firstNameInput, firstName);
        await FillElementAsync(_lastNameInput, lastName);
        await FillElementAsync(_postalCodeInput, postalCode);
    }

    public async Task ContinueToOverviewAsync()
    {
        await ClickElementAsync(_continueButton);
    }

    public async Task CancelCheckoutAsync()
    {
        await ClickElementAsync(_cancelButton);
    }

    public async Task<bool> IsCheckoutPageLoadedAsync()
    {
        return await IsElementVisibleAsync(_checkoutContainer);
    }
}

